Actually fail if ctest fails to verify that all of the kernels work

ryanvolz opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

ryanvolz committed:
Run volk_profile during testing to verify that all of the kernels work.

ryanvolz committed:
MNT: Re-rendered with conda-build 3.21.4, conda-smithy 3.14.3, and conda-forge-pinning 2021.11.11.16.34.59

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

ryanvolz committed:
Fail on ctest failures, skipping known errors on linux-ppc64le.

Since we're actually testing all of the relevant kernels with ctest now,
we can remove the volk_profile test.

ryanvolz committed:
Un-bump build number since package won't be changing after last commit.

ryanvolz commented:
For known failing tests on linux-ppc64le, see https://github.com/gnuradio/volk/issues/442.
